libbeat/processors/add_kubernetes_metadata/matchers.go

```diff
@@ -173,3 +177,94 @@ func (f *FieldFormatMatcher) MetadataIndex(event common.MapStr) string {
 
 	return string(bytes)
 }
+
+// PidMatcher implements matcher that matches container id based on pid from cgroup file
+type PidMatcher struct {
+	matchRegex  string
+	pidCidCache *common.Cache
+}
+
+// NewPidMatcher initializes and returns a PidMatcher
+func NewPidMatcher(cfg common.Config) (Matcher, error) {
+	config := struct {
+		RegexField string `config:"matcher_regex"`
+	}{}
+
+	err := cfg.Unpack(&config)
+	if err != nil {
+		return nil, fmt.Errorf("fail to unpack the `matcher_regex` configuration: %s", err)
+	}
+
+	if len(config.RegexField) == 0 {
+		// no matcher regex from conf, use default one
+		config.RegexField = "^\\d+:\\w+:\\/.+\\/.+\\/.+\\/([0-9a-f]{64})"
+	}
+
+	cache := common.NewCache(time.Minute, 1000)
+	cache.StartJanitor(time.Minute)
+
+	return &PidMatcher{matchRegex: config.RegexField, pidCidCache: cache}, nil
+}
+
+// MetadataIndex returns index for matching metadata indexed based on container id
+func (p *PidMatcher) MetadataIndex(event common.MapStr) string {
+	val, err := event.GetValue("process.pid")
```

Comment:
Specifying which field contains process ID would give more freedom to the user to specify if you want to look at `ppid` or `pid`

Reply:
Please explain more. I'm expecting pid and ppid to be in ratrher concrete place in event. Do you mean that instead of trying with ppid when pid not successful you'd like to see configurable option to try with ppid only?

Reply:
both, with hardcoded `process.pid` and/or `process.ppid` people can't inspect e.g. `system.process.pid`.
For example `add_docker_metadata` allows you to pick which field to inspect for a process id

Reply:
OK. Will make them fields configurable.

Reply:
Done